src/rustup.rs

```diff
@@ -0,0 +1,142 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use anyhow::{anyhow, Error};
+
+use crate::config::Channel;
+use crate::{run, Context};
+
+impl Context {
+    /// Promote a `rustup` release
+    ///
+    /// The [release process] for `rustup` involves copying existing artifacts from one S3 bucket to
+    /// another, updating the manifest, and archiving the artifacts for long-term storage.
+    ///
+    /// `rustup` uses different branches to manage releases. Whenever a commit is pushed to the
+    /// `stable` branch in [rust-lang/rustup], GitHub Actions workflows build release artifacts and
+    /// copy them into `s3://dev-static-rust-lang-org/rustup/dist/`.
```

Comment:
This is not great, I'd prefer if we tweak rustup's CI to upload them to a location like `s3://rustup-artifacts/${commit}`, check what is the latest commit hash on the stable branch, and download from that location. Every build overriding the previous build feels iffy.

Reply:
The bucket has been created [here](https://github.com/rust-lang/simpleinfra/pull/428) and a PR is open [here] to upload artifacts to `s3://rustup-builds/builds/${commit}/`.